pdf/pdfrenderthread.cpp

```diff
@@ -483,7 +483,12 @@ void PDFRenderThreadQueue::processPendingJob()
     case PDFJob::LoadDocumentJob:
         d->loadFailure = false;
         break;
+    case PDFJob::UnLockDocumentJob:
+        job->m_document = d->document;
+        break;
     default:
+        if (d->document->isLocked())
+            return;
```

Comment:
Well, this is dirty fallback solution for the following issue:
- load a valid PDF file and render it;
- load a locked PDF file with the same PdfDocument instance (by calling setDocument());
- having a race condition in PdfCanvas that may call updatePaintNode() from the rendering thread before the page count is reset to zero by line 758.
In that situation you may have some rendering job in the queue before the document is unlocked, which segfault later in pdfjob.cpp#61 when calling page() on a non-existing one.

What do you think ? Is it better to try to cure this race condition, that doesn't happen in practice because we're not reusing the same PdfDocument instance each time ; or should I add a comment here mentioning the reason ?

Besides, writing this, I notice that I should convert the Q_ASSERT() I've added in pdfjob.cpp#61 as a return statement for the same race condition reason with a document smaller than the previous one.